### PR TITLE
- add protable-file-dialogs/0.1.0

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **protable-file-dialogs/0.1.0**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/portable-file-dialogs/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class PortableFileDialogsConan(ConanFile):
+    license = "WTFPL"
```
Comment (danimtb) [inline recipes/portable-file-dialogs/all/conanfile.py:6]:
```suggestion
    name = "portable-file-dialogs"
    license = "WTFPL"
```

File: recipes/portable-file-dialogs/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    def package(self):
+        self.copy("portable-file-dialogs.h", dst="include", src=self._source_subfolder)
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.libdirs = []
```
Comment (danimtb) [inline recipes/portable-file-dialogs/all/conanfile.py:29]: needed? does the hook complain about this?

File: recipes/portable-file-dialogs/all/test_package/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package"
```
Comment (danimtb) [inline recipes/portable-file-dialogs/all/test_package/conanfile.py:7]: cmake_find_package generator is not used
```suggestion
    generators = "cmake"
```

**Review by danimtb — CHANGES_REQUESTED** [review]

File: recipes/portable-file-dialogs/all/conanfile.py (hunk 2 above)
Comment (SSE4) [reply recipes/portable-file-dialogs/all/conanfile.py:30]: I had a warning from conan itself

Comment (SSE4) [reply recipes/portable-file-dialogs/all/conanfile.py:30]:
```
portable-file-dialogs/0.1.0: WARN: Lib folder doesn't exist, can't collect libraries: C:\Users\SSE4\.conan\data\portable-file-dialogs\0.1.0\_\_\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\lib
[HOOK - conan-center.py] post_package_info(): WARN: [LIBRARY DOES NOT EXIST (KB-H054)] Component portable-file-dialogs::portable-file-dialogs libdir "lib" does not contain any declared library
```

File: recipes/portable-file-dialogs/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+from conans import ConanFile, CMake, tools
+
+
+class PortableFileDialogsConan(ConanFile):
+    name = "portable-file-dialogs"
+    license = "WTFPL"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/samhocevar/portable-file-dialogs"
+    description = "Portable GUI dialogs library, C++11, single-header"
+    topics = ("conan", "gui", "dialogs")
```
Comment (uilianries) [inline recipes/portable-file-dialogs/all/conanfile.py:11]:
```suggestion
    topics = ("conan", "gui", "dialogs")
    no_copy_source = True
```

Comment (uilianries) [reply recipes/portable-file-dialogs/all/conanfile.py:11]: Dont need to copy sources

**Review by uilianries — DISMISSED** [review]

**Review by czoido — DISMISSED** [review]

File: recipes/portable-file-dialogs/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import os
+from conans import ConanFile, CMake, tools
```
Comment (SpaceIm) [inline recipes/portable-file-dialogs/all/conanfile.py:2]:
```suggestion
from conans import ConanFile, tools
```

File: recipes/portable-file-dialogs/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/samhocevar/portable-file-dialogs"
+    description = "Portable GUI dialogs library, C++11, single-header"
+    topics = ("conan", "gui", "dialogs")
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
```
Comment (SpaceIm) [inline recipes/portable-file-dialogs/all/conanfile.py:17]:
```suggestion

    def configure(self):
        if self.settings.compiler.get_safe("cppstd"):
            tools.check_min_cppstd(self, 11)

```

File: recipes/portable-file-dialogs/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+class PortableFileDialogsConan(ConanFile):
+    name = "portable-file-dialogs"
+    license = "WTFPL"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/samhocevar/portable-file-dialogs"
+    description = "Portable GUI dialogs library, C++11, single-header"
+    topics = ("conan", "gui", "dialogs")
+    no_copy_source = True
+
```
Comment (SpaceIm) [inline recipes/portable-file-dialogs/all/conanfile.py:13]:
```suggestion
    settings = "compiler"

```

File: recipes/portable-file-dialogs/all/conanfile.py (hunk 6 above)
Comment (SSE4) [reply recipes/portable-file-dialogs/all/conanfile.py:17]: is it really needed? it works without this

**Review by SpaceIm — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Review by mathbunnyru — APPROVED** [review]

**Comment by prince-chrismc** [comment]
FYI, There's a blocking review on this PR

PR | By | Recipe | Reviews | 🛑 Blockers | 🌟 Approvers
-- | -- | -- | -- | -- | --
#4856 | SSE4 | 🆕 portable-file-dialogs | 15 | @danimtb | SpaceIm, uilianries, jgsogo, czoido, mathbunnyru

**Review by uilianries — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Pull request merged**